sql/catalyst/src/main/scala/org/apache/spark/sql/exceptions/SqlScriptingException.scala

```diff
@@ -26,7 +26,7 @@ import org.apache.spark.sql.exceptions.SqlScriptingException.errorMessageWithLin
 class SqlScriptingException (
     errorClass: String,
     cause: Throwable,
-    origin: Origin,
+    val origin: Origin,
```

Comment:
nit: any specific reason for this change? if it's not really important, we usually don't change such minor things

Reply:
`SqlScriptingParserSuite` line 216/217. Now we have to get the line number through origin to check if it is correct. I agree that it would be good if we can avoid this.

Reply:
@miland-db yeah that is the reason, i don't like it either, if you have a suggestion for a better way I'll change it

Reply:
after all, seems fine to leave it as is... the only other option would be to move `val origin...` from primary constructor to the class body, but that would require marking the primary constructor as private and creating new one that would accept `origin` as a param - I would say it's too complicated and basically pretty much the same...

Reply:
Not related to this PR but I feel like that we should use a `QueryContext` instead of `Origin`. @davidm-db @miland-db Have you considered to use the `QueryContext` interface? If so, why refused it.

Reply:
https://github.com/apache/spark/pull/47462#discussion_r1714418202
Does this sound fine to you? Any suggestions on how to use `QueryContext` for the same purpose?

Reply:
I feel we should review this decision more carefully. @srielau @cloud-fan  WDYT?

Reply:
can't we add `def lineNumber: Int` in this exception? `Origin` is not a public API and shouldn't be exposed to end users.